src/encord_active/lib/versioning/git.py

```diff
@@ -0,0 +1,73 @@
+from pathlib import Path
+from typing import Literal, NamedTuple, Optional, Union
+
+from git.objects import Commit
+from git.repo import Repo
+
+GITIGNORE = """data/**
+embeddings/**
+**/*.zip
+"""
+
+
+class Version(NamedTuple):
+    name: str
+    id: str
+
+
+class GitVersioner:
+    def __init__(self, path: Path) -> None:
+        if (path / ".git").exists():
+            self.repo = Repo(path)
+        else:
+            self.repo = Repo.init(path)
+            with self.repo.config_writer() as writer:
+                writer.set_value('diff "sqlite3"', "textconv", '"f() { sqlite3 \\"$@\\" .dump; }; f"')
+            (path / ".git/info/attributes").write_text("sqlite.db diff=sqlite3")
+            (path / ".gitignore").write_text(GITIGNORE)
+            self.repo.index.add(self.repo.untracked_files)
+            self.repo.index.commit("init")
+
+    @property
+    def current_version(self):
+        return _commit_to_version(self.repo.head.commit)
+
+    def is_latest(self, version: Optional[Version] = None) -> bool:
+        return (version or self.current_version).id == self.repo.heads.main.commit.hexsha
+
+    @property
+    def versions(self):
+        return [_commit_to_version(commit) for commit in self.repo.iter_commits(self.repo.heads.main)]
+
+    @property
+    def has_changes(self):
+        return self.repo.is_dirty() or bool(self.repo.untracked_files)
+
+    def create_version(self, name: str):
```

Comment:
Since we expose this via the `lib`, should we raise an error if we're not on the latest version?

Reply:
Yep, sounds like a good idea :+1: